starter-code/src/functions-and-arrays.js

```diff
@@ -11,14 +20,70 @@ var words = [
   'crackpot'
 ];
 
+function findLongestWord (words) {
+  if (words.length === 0) {
+    return undefined;
+  }
+  var longestWord = '';
+  for(var i = 0; i < words.length; i++){
+    if(words[i].length > longestWord.length){
+      longestWord = words[i];
+    }
+  }
+  
+  return longestWord;
+  
+}
+findLongestWord(words)
+
 // Calculating a Sum
 
 var numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
+//Suma sin reduce
+
+function sumArray (arrayNumbers) {
+  if (arrayNumbers.length === 0) {
+    return 0;
+  }
+  var accumulator = 0;
+  for(var i = 0; i < arrayNumbers.length; i++){
+    accumulator = accumulator + arrayNumbers[i]
+  }
+  return accumulator;
+}
+sumArray(numbers);
+
+//Suma con reduce
+/*
+function sumArray(numbers) {
+  if(numbers.length === 0) {
+    return 0;
+  }
+  var numSum = numbers.reduce((prev, cur) => prev + cur, 0);
+  return numSum;
+}
+
+sumArray(numbers);
+*/
+
 // Calculate the Average
 
 var numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
+function averageNumbers (arrayNumbers) {
+  if (arrayNumbers.length === 0) {
+    return undefined;
+  }
+  var accumulator = 0;
+  for (var i = 0; i < arrayNumbers.length; i++) {
+    accumulator = accumulator + arrayNumbers[i];
+  }
+  return  accumulator / arrayNumbers.length;
+}
```

Comment:
Muy bien, pero fijaos, es igual que el ejercicio anterior más una división, lo podéis hacer con reduce( ) también